tesseract_common/include/tesseract_common/plugin_loader.h

```diff
@@ -118,6 +118,19 @@ class PluginLoader
    * @return The number of plugins.
    */
   inline int count() const;
+
+  /**
+   * @brief Utility function to add library containing symbol to the search env variable
+   * 
+   * In some cases the name and location of a library is unknown at runtime, but a symbol can
+   * be linked at compile time. This is true for Python auditwheel distributions. This
+   * utility function will determine the location of the library, and add it to the library search
+   * environment variable so it can be found.
+   * 
+   * @param symbol_ptr Pointer to the symbol to find
+   * @param search_libraries_env The environmental variable to modify
+   */
+  static inline void addSymbolLibraryToSearchLibrariesEnv(const void* symbol_ptr, const std::string& search_libraries_env);
```

Comment:
This should also update the environment variable for search paths.
``` c++
void addSymbolLibraryToSearchLibrariesEnv(const void* symbol_ptr, const std::string& search_libraries_env, const std::string& search_paths_env)
{
  std::string search_libraries_env_var_str;
  char* search_libraries_env_var = std::getenv(search_libraries_env.c_str());
  if (search_libraries_env_var != nullptr)
    search_libraries_env_var_str = search_libraries_env_var;

  std::string search_paths_env_var_str;
  char* search_paths_env_var = std::getenv(search_paths_env.c_str());
  if (search_paths_env_var != nullptr)
    search_paths_env_var_str = search_paths_env_var;

  boost::dll::fs::path lib_path = boost::dll::symbol_location_ptr(symbol_ptr);

  std::string separator {":"};
#ifdef _WIN32
  separator = ";";
#endif

  if (search_libraries_env_var_str.empty())
    search_libraries_env_var_str = lib_path.filename().string();
  else
    search_libraries_env_var_str = search_libraries_env_var_str + separator + lib_path.filename().string();

  if (search_paths_env_var_str.empty())
    search_paths_env_var_str = lib_path.parent_path().string();
  else
    search_paths_env_var_str = search_paths_env_var_str + separator + lib_path.parent_path().string();

#ifndef _WIN32
  setenv(search_libraries_env.c_str(),search_libraries_env_var_str.c_str(),1);
  setenv(search_paths_env.c_str(),search_paths_env_var_str.c_str(),1);
#else
  _putenv_s(search_libraries_env.c_str(),search_libraries_env_var_str.c_str());
  _putenv_s(search_paths_env.c_str(),search_paths_env_var_str.c_str());
#endif
}
```

Reply:
The library environment variable should only contain the library name and not the full path.

Reply:
For this to work I also had to update the decorate function to check if an extension already exists before adding a suffix.
``` c++
  if (actual_path.extension().empty())
    actual_path += boost::dll::shared_library::suffix();
```

Reply:
It is a lot safer to use absolute paths for the library loading with this method. Otherwise it may wind up loading an unexpected library it finds somewhere else in the search paths.

Reply:
Paths provided by the environment variable are search first for this reason.

Reply:
It may be good to have this method prepend  instead of append to the environment variable.

Reply:
I am not sure what the issue is with the library name or absolute path, it is working fine with the test I added.

Reply:
Could we add another environmental variable or a static field that has absolute paths?

Reply:
I guess we could change it to loop over the libraries and check for full path. If full path these get searched first which makes sense to me.

Reply:
Yeah, that should work. I also added logic to check for `:` on Linux and `;` on Windows. This is an annoying difference between operating systems since `:` can be part of a path on Windows.

Reply:
@marip8 What do you think here?

Reply:
@johnwason I updated it to support libraries provided as full path. The changes can be seen if you follow the line to the PR below. We are waiting on this to be released on the build farm to switch this repository over to using.

Reply:
@Levi-Armstrong are you going to backport your updates or should I include them here?

Reply:
I think it would be good to back port them here this way to get things merged in here for now. We are going to make some changes it boost_plugin_loader to better support this functionality but may be a week or so before it gets merged in.